src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -1306,3 +1393,69 @@ int sd_dhcp_server_set_relay_target(sd_dhcp_server *server, const struct in_addr
         server->relay_target = *address;
         return 1;
 }
+
+int sd_dhcp_server_add_static_lease(
+                sd_dhcp_server *server,
+                const struct in_addr* address,
+                uint8_t *mac_addr,
+                size_t mac_addr_size) {
+         _cleanup_(dhcp_lease_freep) DHCPLease *lease = NULL, *old = NULL;
+        int r;
+
+        assert_return(server, -EINVAL);
+        assert_return(address, -EINVAL);
+        assert_return(mac_addr, -EINVAL);
+        assert_return(mac_addr_size == ETH_ALEN, -EINVAL);
```

Comment:
Please refuse to set new static lease when the server is always running, as the address may be already assigned to a host.

Reply:
@yuwata  please check if current changes reflect on what You had in mind.